File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    var answer = 1;
    let sum=0;
    for(let i =1;i<n/2;i++){
        sum +=i;
        for(let j =i+1;j<n;j++){
            sum +=j;
            if(sum ==n){ answer++;}
            else if(sum>n){break;}
        }
        sum =0;
    }
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    left, right, s = 1, 1, 1
    
    while left <= n:
        if s == n:
            answer += 1
            s -= left
            left += 1
        elif s < n:
            right += 1
            s += right
        else:  # s > n
            s -= left
            left += 1
            
    return answer
